packages/slice-machine/src/components/Card/Card.tsx

```diff
@@ -0,0 +1,108 @@
+import { Text } from "@prismicio/editor-ui";
+import { clsx } from "clsx";
+import {
+  type CSSProperties,
+  type FC,
+  type HTMLAttributes,
+  type ImgHTMLAttributes,
+  type MouseEvent,
+  type PropsWithChildren,
+  type ReactNode,
+  createElement,
+} from "react";
+
+import * as styles from "./Card.css";
+
+type CardProps = PropsWithChildren<
+  {
+    size?: keyof typeof styles.size;
+    style?: CSSProperties;
+    variant?: keyof typeof styles.variant;
+  } & (
+    | {
+        interactive: true;
+        checked?: boolean;
+        disabled?: boolean;
+        onClick?: (event: MouseEvent) => void;
+      }
+    | {
+        interactive?: false;
+        checked?: never;
+        disabled?: never;
+        onClick?: never;
+      }
+  )
+>;
+
+export const Card: FC<CardProps> = (props) => {
+  const {
+    checked: _checked,
+    disabled: _disabled,
+    interactive: _interactive,
+    onClick: _onClick,
+    size = "medium",
+    variant = "solid",
+    ...otherProps
+  } = props;
+  return (
+    <div
+      {...otherProps}
+      className={clsx(styles.root, styles.size[size], styles.variant[variant], {
+        [styles.interactive]: props.interactive,
+        [styles.interactiveVariant[variant]]: props.interactive,
+      })}
+      // TODO: add missing ARIA attributes and keyboard event handlers.
+      {...(props.interactive === true
+        ? {
+            "data-disabled": props.disabled === true ? "" : undefined,
+            "data-state": props.checked === true ? "checked" : undefined,
+            onClick: props.disabled === true ? undefined : props.onClick,
+            tabIndex: props.disabled === true ? undefined : 0,
+          }
+        : {})}
+    />
+  );
+};
+
+type CardMediaProps =
+  | ({ component: "div" } & HTMLAttributes<HTMLDivElement>)
+  | ({ component?: "img" } & ImgHTMLAttributes<HTMLImageElement>);
+
+export const CardMedia: FC<CardMediaProps> = ({
+  className,
+  component = "img",
+  ...otherProps
+}) =>
+  createElement(component, {
+    ...otherProps,
+    className: clsx(styles.media, styles.mediaComponent[component], className),
+  });
+
+export const CardActions: FC<PropsWithChildren> = (props) => (
+  <div {...props} className={styles.actions} />
+);
+
+type CardFooterProps = {
+  action?: ReactNode;
+  subtitle?: ReactNode;
+  title?: ReactNode;
+};
+
+export const CardFooter: FC<CardFooterProps> = ({
+  action,
+  subtitle,
+  title,
+  ...otherProps
```

Comment:
❓ I'm always wondering, here you are typing the props with strictly 3 props. But you spread the rest. It should have no rest, at least it's not visible it's allowed. Should we extend the type to explain we accept some other HTML default props?